Function/BookFunc/Services/BookingService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+using Microsoft.Graph.Auth;
+using TeamsMeetingBookFunc.Models;
+using TeamsMeetingBookFunc.Helpers;
+using Microsoft.Azure.WebJobs;
+
+namespace TeamsMeetingBookFunc.Services
+{
+    class BookingService
+    {
+        internal IConfigurationRoot Configuration { get; private set; }
+
+        internal GraphServiceClient GraphServiceClient { get; private set; }
+
+        #region Lazy and thread-safe singleton
+        private static readonly Lazy<BookingService> _current = new Lazy<BookingService>(() => new BookingService());
+
+        internal static BookingService Current => _current.Value;
+
+        private BookingService()
+        {
+
+        }
+        #endregion
+
+        internal void Init(ExecutionContext context){
+
```

Comment:
What if someone calls Init many times? Should we have an "initialized" bool variable to check?